### PR TITLE
bpo-40280: WASM docs and smaller browser builds (GH-32412)

### PR DESCRIPTION
**Pull request by tiran**
https://bugs.python.org/issue40280

### CODE REVIEW

File: Tools/wasm/README.md [hunk 1]
```diff
@@ -9,9 +9,21 @@ possible to build for *wasm32-wasi* out-of-the-box yet.
 
 ## wasm32-emscripten build
 
-Cross compiling to wasm32-emscripten platform needs the [Emscripten](https://emscripten.org/)
-tool chain and a build Python interpreter.
-All commands below are relative to a repository checkout.
+Cross compiling to wasm32-emscripten platform needs the
```
Comment (brettcannon) [inline Tools/wasm/README.md]:
```suggestion
Cross compiling to the wasm32-emscripten platform needs the
```

File: Tools/wasm/README.md [hunk 2]
```diff
@@ -167,3 +179,77 @@ linker options.
 - pthread support requires WASM threads and SharedArrayBuffer (bulk memory).
   The runtime keeps a pool of web workers around. Each web worker uses
   several file descriptors (eventfd, epoll, pipe).
+
+# Hosting Python WASM builds
+
+The simple REPL terminal uses SharedArrayBuffer. For security reasons
+browsers only provide the feature in secure environents with cross-origin
+isolation. The webserver must sent cross-origin headers and correct mimetypes
```
Comment (brettcannon) [inline Tools/wasm/README.md]:
```suggestion
isolation. The webserver must send cross-origin headers and correct MIME types
```

File: Tools/wasm/README.md [hunk 3]
```diff
@@ -167,3 +179,77 @@ linker options.
...
+for the JavaScript and WASM files. Otherwise the terminal will fail to load
+an error message like ``Browsers disable shared array buffer``.
```
Comment (brettcannon) [inline Tools/wasm/README.md]:
```suggestion
with an error message like ``Browsers disable shared array buffer``.
```

File: Tools/wasm/wasm_assets.py [hunk 4]
```diff
@@ -105,34 +122,59 @@
...
+def detect_extension_modules(args: argparse.Namespace):
+    modules = {}
+
+    # disabled by Modules/Setup.local ?
+    with open(args.builddir / "Makefile") as f:
+        for line in f:
+            if line.startswith("MODDISABLED_NAMES="):
+                disabled = line.split("=", 1)[1].strip().split()
+                for modname in disabled:
+                    modules[modname] = False
```
Comment (brettcannon) [inline Tools/wasm/wasm_assets.py:158]: Any reason this isn't a set?

**Review by brettcannon — APPROVED** [review]

File: Tools/wasm/wasm_assets.py (hunk 4 above)
Comment (tiran) [reply Tools/wasm/wasm_assets.py:158]: I have to track disabled and later enabled modules. A dict seemed easier to use than a set.

**Pull request merged**